be/src/storage/tablet_schema.h

```diff
@@ -163,6 +163,12 @@ class TabletColumn {
     void add_sub_column(TabletColumn&& sub_column);
     uint32_t subcolumn_count() const { return _extra_fields ? _extra_fields->sub_columns.size() : 0; }
     const TabletColumn& subcolumn(uint32_t i) const { return _extra_fields->sub_columns[i]; }
+    const TabletColumn* subcolumn_ptr(uint32_t i) const {
```

Comment:
> I see that there is a function `TabletColumn& subcolumn(uint32_t i)`. How do you avoid that some developers will call that function with a non-exist field ID? If this happened, the process will crash. And what does the input argument `i` mean? Does it mean the ordinal or the field_id in the thrift?

`i` means the ordinal index of sub column. And if `i` is invalid, it will return nullptr.

Reply:
I mean the above function `subcolumn`, not this new function.

Reply:
> function

I believe the `subcolumn` function cannot ensure safety when i is invalid. Currently, the function is called with the guarantee that i is valid. It might be better to add some safety checks, I will add some safety check.

Reply:
why not remove this function and change all the callers to use this new functions and check if the returned value is null or not.

Reply:
> why not remove this function and change all the callers to use this new functions and check if the returned value is null or not.

I think this change can be implemented in a new PR because it involves quite a few modifications. Currently, the callers of the function `subcolumn` assume the index is valid, so many places do not implement exception handling logic. Not only would exception handling logic need to be added for the callers, but also for the higher-level callers. The entire call chain must ensure proper exception handling, otherwise, it could still cause BE crash. These modifications are significant, so I don't think it is appropriate to include them in this PR.